### DISCUSSION TITLE
Why is a Single window referred to as "WindowType::Normal"?

### DISCUSSION THREAD

**Discussion by blueglyph** [fltk-rs/fltk-rs#1235] [Q&A] [answered]
I saw something strange when fl2rust tried to translate a fluid *.fl file with a single window (instead of a double-buffered window).

It translated `Fl_Window my_win` to `my_win.set_type(WindowType::Single);`, which doesn't work because in this crate, we have (in windows.rs):

```rust
pub enum WindowType {
    /// Single window
    Normal = 240,
    /// Double window
    Double = 241,
}
```

If I manually change the generated *.rs file to `my_win.set_type(WindowType::Normal);`, then it compiles and runs as expected.

I was wondering why it was named `Normal` instead of `Single`? And should the `fl2rust` code be modified to take this into account, or should this crate use `Single` instead of `Normal`. The latter seems to be the best option, since it more explicitly refers to the single-buffered type. "Normal" is likely to be the double-buffering option from what I read, but I may be wrong.

The problem with either choice is the version increment due to non-compatibility with previous code, but a fix is required either way.

**Comment by MoAlyousef** [comment, marked as answer]
It's a vestige of the first code in fltk-rs that I forgot to change. The problem was in the Fl_Window.H header, it's described as:
```cpp
#define FL_WINDOW 0xF0          ///< window type id: all subclasses have type() >= this
#define FL_DOUBLE_WINDOW 0xF1   ///< double window type id
```
I've added an alias WindowType::Single, and a roadmap (to version2) point to remove WindowType::Normal. I've also fixed the fl2rust to generate WIndowType::Normal for single-buffered windows, and when version2 is released, to generate WindowType::Single.

**Reply by blueglyph** [reply]
Thanks! I preferred to ask here before submitting a bug since I was not sure where this was coming from, or if I had done something wrong.
